src/batch-client/index.ts

```diff
@@ -433,6 +436,57 @@ export class ZimbraBatchClient {
 			return ids ? ids.split(',') : [];
 		});
 
+	// For offline Drafts
+	public attach = (files: any, { ...message }) => {
+		const promises = castArray(files).map(file => this.localStoreClient.attach({ file }));
+
+		return Promise.all(promises).then(attached => {
+			// After all uploads have completed, add attachments to the message and save it as a draft.
+			attached.forEach((attachment, index) => {
+				if (attachment) {
+					const file = files[index];
+					if (attachment.contentDisposition === 'inline') {
+						const { base64, url, part, messageId } = attachment;
+						const contentType = file.type || file.contentType;
+						const fileAttributes = {
+							filename: file.name || file.filename,
+							contentType,
+							size: file.size,
+							contentId: file.contentId,
+							contentDisposition: file.contentDisposition,
+							...(part && { part }),
+							...(messageId && { messageId }),
+							...(base64 && { base64 }),
+							...(url
```

Comment:
looks like we are merging properties from file and attachment object, can we merge using destructure and then update some properties?

Reply:
Code refactored and moved to zm-x-web